examples/compiler/01_lowering.ipynb

```diff
@@ -0,0 +1,347 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Lowering\n",
+    "\n",
+    "In the first compilation pass, three main tasks are carried out: _indexification_, _substitution_, and _domain-alignment_.\n",
+    "\n",
+    "\n",
+    "### Indexification\n",
+    "\n",
+    "The _indexification_ consists of converting Functions into Indexeds, that is actual array accesses. An Indexed always keeps a reference to its originating function. For instance, all accesses to `u` such as `u[t, x + 1]` and `u[t + 1, x − 2]`would store a pointer to the same, user-defined `Function` _u(t, x)_. This metadata is exploited throughout the various compilation passes. \n",
+    "\n",
+    "To see this, let's create a `TimeFunction` named `u`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "u(time, x)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from devito import Grid, TimeFunction\n",
+    "u = TimeFunction(name='u', grid=Grid((3,)), save=1)\n",
+    "u"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that `u` has one temporal dimension and one spatial dimension, that we will call `t` and `x` here, respectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(time, x)"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "t,x = u.dimensions\n",
```

Comment:
again, probably should be ``time, x = ...``

Reply:
ok